src/language/dafnyInstallation.ts

```diff
@@ -24,32 +25,53 @@ function ifNullOrEmpty(a: string | null, b: string): string {
   return a === null || a === '' ? b : a;
 }
 
-function getConfiguredVersion(): string {
-  const version = Configuration.get<string>(ConfigurationConstants.PreferredVersion);
-  return version === LanguageServerConstants.Latest
-    ? LanguageServerConstants.LatestVersion
-    : version;
+async function getConfiguredVersion(): Promise<string> {
+  const [ _, version ] = await getConfiguredTagAndVersion();
+  return version;
+}
+
+let getConfiguredTagAndVersionCache: [string, string];
+async function getConfiguredTagAndVersion(): Promise<[string, string]> {
+  return getConfiguredTagAndVersionCache ?? (getConfiguredTagAndVersionCache = await getConfiguredTagAndVersionUncached());
+}
+
+async function getConfiguredTagAndVersionUncached(): Promise<[string, string]> {
+  let version = Configuration.get<string>(ConfigurationConstants.PreferredVersion);
+  switch(version) {
+  case LanguageServerConstants.LatestStable:
+    version = LanguageServerConstants.LatestVersion;
+    break;
+  case LanguageServerConstants.LatestNightly: {
+    const result: any = await (await fetch('https://api.github.com/repos/dafny-lang/dafny/releases/tags/nightly')).json();
+    if(result.name !== undefined) {
+      const version = result.name.substring(6);
```

Comment:
```suggestion
      const version = result.name.substring("Dafny ".length);
```

Perhaps even with a sanity `startsWith` check beforehand :)